Connect the frontend application form to the backend

chrislessard opened the pull request:
Hi,

This PR introduces chances that will allow the application form content to be posted to the backend server, thus effectively allowing applicants to submit the application. 

Solves #67. Also reliant on PR https://github.com/uoft-tapp/tapp/pull/95, which resolves a bug that breaks the build.

freeatnet commented on app/services/chass_importer.rb:
```diff
@@ -31,23 +31,31 @@ def get_round_id
-      Rails.logger.debug db_model.attributes
+  def insertion_helper(model, data, ident, exists)
+    unless model.where(ident).exists?
+      db_model = model.create(data)
```
We can use [`create!`](http://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-create-21) here to create the record and raise an exception.

michellemtchai replied:
@freeatnet Should I add this in a new PR?

freeatnet replied:
@michellemtchai I think this is something that can be put on the "when cleaning up" list :)

freeatnet commented on app/services/chass_importer.rb:
```diff
@@ -31,23 +31,31 @@ def get_round_id
       Rails.logger.debug "#{exists}"
+      db_model = model.find_by(ident)
+      Rails.logger.debug "existing model #{JSON.pretty_generate(db_model.as_json)}\n"
+      db_model.update_attributes(data)
```
Same here, [`update_attributes!`](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update-21) would work

chrislessard committed:
fix(pre-commit): Update version of eslint

This patch ammends a bug which used a newer version of eslint,
rather than the one included as part of react-scripts, causing the
app to crash.

chrislessard committed:
feat(frontend): Connect application form to backend

This patch introduces code that will take application form data
and post it to the backend server.

chrislessard committed:
enhancement(frontend): Add better messages to application POST

This patch introduces more detailed error messages on the frontend
when POSTing an application to the backend server.
